src/dibs.cpp

```diff
@@ -1,11 +1,74 @@
+#include <detail/glfw_instance.hpp>
+#include <detail/log.hpp>
 #include <dibs/dibs.hpp>
 #include <dibs/dibs_version.hpp>
+#include <cassert>
 
 namespace dibs {
-struct Instance::Impl {};
+namespace {
+struct Glfw {
+	detail::UniqueGlfw instance;
+	detail::UniqueWindow window;
+};
 
-Instance::Instance() : m_impl(std::make_unique<Impl>()) {}
+Result<Glfw> makeGlfw(char const* title, uvec2 const extent, Instance::Flags const flags) noexcept {
+	if (detail::g_window) { return Error::eDuplicateInstance; }
+	if (extent.x == 0U || extent.y == 0U) { return Error::eInvalidExtent; }
+	auto instance = detail::GlfwInstance::make();
+	if (!instance) { return Error::eGlfwInitFailure; }
+	if (!glfwVulkanSupported()) { return Error::eUnsupportedPlatform; }
+	auto window = instance->makeWindow(title, extent, flags);
+	if (!window) { return Error::eWindowCreationFailure; }
+	return Glfw{std::move(instance), std::move(window)};
+}
+
+bool centre(GLFWwindow* window) {
+	auto const monitor = glfwGetPrimaryMonitor();
+	if (!monitor) { return false; }
+	auto const mode = glfwGetVideoMode(monitor);
+	if (!mode) { return false; }
+	int w{}, h{};
+	glfwGetWindowSize(window, &w, &h);
+	if (w <= 0 || h <= 0) { return false; }
+	glfwSetWindowPos(window, (mode->width - w) / 2, (mode->height - h) / 2);
+	return true;
+}
+} // namespace
+
+using Clock = std::chrono::steady_clock;
+
+struct Instance::Impl {
+	Glfw glfw;
+	Clock::time_point elapsed = Clock::now();
+};
+
+Instance::Instance(std::unique_ptr<Impl>&& impl) noexcept : m_impl(std::move(impl)) {}
 Instance::Instance(Instance&&) noexcept = default;
 Instance& Instance::operator=(Instance&&) noexcept = default;
-Instance::~Instance() noexcept = default;
+Instance::~Instance() noexcept {
+	if (m_impl) { detail::g_window = {}; }
+}
+
+bool Instance::closing() const noexcept {
+	assert(m_impl);
+	return glfwWindowShouldClose(m_impl->glfw.window);
+}
+
+Time Instance::poll() noexcept {
+	assert(m_impl);
+	glfwPollEvents();
+	auto const t = Clock::now();
+	return t - std::exchange(m_impl->elapsed, t);
```

Comment:
Can you explain why'd we need `std::exchange` here I read the entry over on cppreference so I can kinda have an idea why you would use it